embedded/devices/src/ads1219.cpp

```diff
@@ -0,0 +1,64 @@
+#include "embedded/devices/include/ads1219.h"
+
+
+#define WRITE_CONFIG_REG 0x40 // 0100 00xx
+#define READ_CONFIG_REG 0x20 // 0010 00xx
+#define READ_STATUS_REG 0x24 // 0010 01xx
+#define RESET 0x06 // 0000 011x
+#define RDATA 0x10 // 0001 xxxx
+#define START_CONVERSION 0x08 // 0000 100x
+#define CONVERSION_VOLTAGE 244.14e-9
+
+Ads1219::Ads1219(int bus, int addr) : I2c(bus, addr, O_RDWR){}
+
+void Ads1219::loop() {
+    while (this->stop_thread != true) {
+        // get the current channel we are on
+        uint8_t configReg = this->read_from_reg(READ_CONFIG_REG); 
+        uint8_t currentChannel = configReg >> 5;
+        // calculate the next channel
+        if(currentChannel == 6 || currentChannel == 0){
+            currentChannel = 3;
+        }
+        else{
+            currentChannel++;
+        }
+        int channel = currentChannel - 3; // used to access the voltages array
+        // set up the next configReg 
+        configReg = (configReg & 0x1F) | (currentChannel << 5);
+        // switch the channel
+        this->write_data<uint8_t>(WRITE_CONFIG_REG, configReg);
+        // read status register until conversion is done(it has a 1 flag)
+        uint8_t statusReg = 0;
```

Comment:
Let's just start with 
```c++
uint8_t statusReg = this->read_from_reg(READ_STATUS_REG)
```
on the off chance that the conversion completes right away, this will save time

Reply:
this won't work because we need to get the leading bit only, so it's best to have the while loop iterate at least once